### ISSUE TITLE
hist / hist2d inconsistent for plotting single data point

### ISSUE THREAD

**Issue by eyurtsev** [matplotlib/matplotlib#3882]
matplotlib version 1.4.2

hist throws an error when fed with a single data point, but hist2d doesn't.

``` python
figure()
hist2d([1], [1]) # works

figure()
hist(1)  # throws a ValueError
```

The error is: "ValueError: x has only one data point. bins or range kwarg must be given"

**Comment by sque** [comment]
I have a different problem but related problem with `hist` function.

matplotlib 1.4.2

```python
x = [1]
plt.hist(x, bins=10)
```
`ValueError: x has only one data point. bins or range kwarg must be given`

The problem here is that although I am explicitly defining the `bins` I still get an error for single data point. So, I am wondering if this patch has fixed this issue, or should I open a new one.

**Comment by tacaswell** [comment]
This case will also be fixed by #3883 , but that is only on the master branch and won't be included in 1.4.3.

What OS are you using?  It might be possible to install from source or to locally apply the patch from #3883.

**Comment by sque** [comment]
I tried 1.4.3rc1 branch and found it still didn't work. My mistake, I should I have noticed that this was not included in 1.4.3.

Just checked out master and this problem has disappeared :) Unfortunately I have to use a custom `hist` function until this fix gets released.

Thanks!
